ignite/contrib/handlers/tqdm_logger.py

```diff
@@ -0,0 +1,92 @@
+try:
+    from tqdm import tqdm
+except ImportError:
+    raise RuntimeError("This contrib module requires tqdm to be installed")
+
+from ignite.engine import Events
+
+
+class ProgressBar:
+    """
+    TQDM progress bar handler to log training progress and computed metrics
+
+    Args:
+        engine (ignite.Engine): an engine object
+        loader (iterable or DataLoader): data loader object
+        output_transform: transform a function that transforms engine.state.output
+                into a dictionary of format {metric_name: metric_value}
+        mode (str): 'iteration' or 'epoch' (default=epoch)
+        log_interval (int or None): interval of which the metrics information is displayed.
+                            If set to None, only the progress bar is shown and not
+                            the metrics. (default=1)
+
+    Example:
+        (...)
+        pbar = ProgressBar(trainer, train_loader, output_transform=lambda x: {'loss': x})
+        trainer.add_event_handler(Events.ITERATION_COMPLETED, pbar)
+
+    Note:
+        1) Bear in mind that `output_transform` should return a dictionary, whose values are floats.
+        This is due to the running average over every metric that is being computed. If you have
+        metrics that are tensors or arrays, you will have to unroll each value to its own
+        dictionary key.
+
+        2) When adding this handler to an engine, it is recommend that you replace every print
+        operation in the engine's handlers with `tqdm.write()` to guarantee the correct stdout format.
+    """
+
+    def __init__(self, engine, loader, output_transform=lambda x: x, mode='epoch', log_interval=1):
+        self.num_iterations = len(loader)
+        self.metrics = {}
+        self.output_transform = output_transform
+        self.pbar = None
+        self.mode = mode
+        self.log_interval = log_interval
+
+        if log_interval is not None:
+            assert log_interval >= 1, 'log_frequency must be positive'
+
+        assert mode in {'iteration', 'epoch'}, \
+            'incompatible mode {}, accepted modes {}'.format(mode, {'iteration', 'epoch'})
+
+        log_event = Events.EPOCH_COMPLETED if mode == 'epoch' else Events.ITERATION_COMPLETED
+
+        engine.add_event_handler(Events.EPOCH_STARTED, self._reset)
+        engine.add_event_handler(Events.EPOCH_COMPLETED, self._close)
+        engine.add_event_handler(log_event, self._log_message)
+
+    def _reset(self, engine):
+        self.pbar = tqdm(
+            total=self.num_iterations,
+            leave=False,
+            bar_format='{desc}[{n_fmt}/{total_fmt}] {percentage:3.0f}%|{bar}{postfix} [{elapsed}<{remaining}]')
+
+    def _close(self, engine):
+        self.pbar.close()
+
+    def _log_message(self, engine):
```

Comment:
What do you think about to provide a public method `log_message` that users can use in their custom event to log something particular. For example,
```
pbar = ...

@trainer.on(...)
def log_validation_scores(engine):
    evaluator.run(validation_dataset)
    metrics = evaluator.state.metrics
    pbar.log_message("Val loss : {}".format(metrics['loss']))

```